commit 77b1374b99ee91d23d8802dd1dec9a5f630e919a
Author: Aaron Seigo <aseigo@kde.org>
Date:   Wed Nov 9 20:15:50 2011 +0100

    only delete the layout when it isn't assigned to another layout
    
    based on the patch by Raul Fernandes
    BUG:275286

diff --git a/plasma/desktop/applets/tasks/taskgroupitem.cpp b/plasma/desktop/applets/tasks/taskgroupitem.cpp
index 6b23b71..d7816d3 100644
--- a/plasma/desktop/applets/tasks/taskgroupitem.cpp
+++ b/plasma/desktop/applets/tasks/taskgroupitem.cpp
@@ -79,8 +79,11 @@ TaskGroupItem::TaskGroupItem(QGraphicsWidget *parent, Tasks *applet)
 
 TaskGroupItem::~TaskGroupItem()
 {
-    delete m_tasksLayout;
-    m_tasksLayout = 0;
+    if (!m_offscreenLayout && !m_mainLayout) {
+        // only delete this if we have neither an offscreen layout or a mainlayout
+        // if we do, then they will delete the layout for us.
+        delete m_tasksLayout;
+    }
 }
 
 bool TaskGroupItem::isSplit()
